zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/archiver/ElasticsearchRepositoryIT.java

```diff
@@ -203,7 +204,75 @@ void shouldReindexDocuments() throws IOException {
   }
 
-  private void index(final String index, final TestDocument document) {
+  @Test
+  void shouldGetProcessInstancesNextBatch() throws IOException {
+    // given - 5 documents, two of which were created over an hour ago, one of which was created,
+    // one which is not a process instance join relation, and one which is not on the right
+    // partition
+    final var now = Instant.now();
+    final var twoHoursAgo = now.minus(Duration.ofHours(2)).toString();
+    final var repository = createRepository();
+    final var documents =
+        List.of(
+            new TestProcessInstance(
+                "1", twoHoursAgo, ListViewTemplate.PROCESS_INSTANCE_JOIN_RELATION, 1),
+            new TestProcessInstance(
+                "2", twoHoursAgo, ListViewTemplate.PROCESS_INSTANCE_JOIN_RELATION, 2),
+            new TestProcessInstance("3", twoHoursAgo, ListViewTemplate.ACTIVITIES_JOIN_RELATION, 1),
+            new TestProcessInstance(
+                "4", now.toString(), ListViewTemplate.PROCESS_INSTANCE_JOIN_RELATION, 1));
+
+    // create the index template first to ensure ID is a keyword, otherwise the surrounding
+    // aggregation will fail
+    createProcessInstanceIndex();
+    documents.forEach(doc -> index(processInstanceIndex, doc));
+    testClient.indices().refresh(r -> r.index(processInstanceIndex));
+    config.setRolloverBatchSize(3);
+
+    // when
+    final var result = repository.getProcessInstancesNextBatch();
+
+    // then - we expect only the first two documents created two days ago to be returned
+    final var dateFormatter =
+        DateTimeFormatter.ofPattern("yyyy-MM-dd").withZone(ZoneId.systemDefault());
+    assertThat(result).succeedsWithin(Duration.ofSeconds(30));
+    final var batch = result.join();
+    assertThat(batch.ids()).containsExactly("1");
+    assertThat(batch.finishDate()).isEqualTo(dateFormatter.format(now.minus(Duration.ofHours(2))));
+  }
+
+  @Test
+  void shouldGetBatchOperationsNextBatch() throws IOException {
+    // given - 3 documents, two of which were created over an hour ago, one of which was created
+    final var now = Instant.now();
+    final var twoHoursAgo = now.minus(Duration.ofHours(2)).toString();
+    final var repository = createRepository();
+    final var documents =
+        List.of(
+            new TestBatchOperation("1", twoHoursAgo),
+            new TestBatchOperation("2", twoHoursAgo),
+            new TestBatchOperation("3", now.toString()));
+
+    // create the index template first to ensure ID is a keyword, otherwise the surrounding
+    // aggregation will fail
+    createBatchOperationIndex();
+    documents.forEach(doc -> index(batchOperationIndex, doc));
+    testClient.indices().refresh(r -> r.index(batchOperationIndex));
+    config.setRolloverBatchSize(3);
+
+    // when
+    final var result = repository.getBatchOperationsNextBatch();
+
+    // then - we expect only the first two documents created two days ago to be returned
```

Comment:
here is hours as well, and same for the test above :)

Reply:
Oops! That was from the previous iteration :see_no_evil: